### PR TITLE
Feature 5

### PR DESCRIPTION
**Pull request by Namrata-J**
#6 in this PR I have added auth using Mockbee.
Please review it, thank you!!

### CODE REVIEW

File: src/components/index.js [hunk 1]
```diff
@@ -9,5 +9,6 @@ import { ArchivedNotes } from "./ArchivedNotes/ArchivedNotes";
 import { DeletedNotes } from "./DeletedNotes/DeletedNotes";
 import { SearchBar } from "./SearchBar/SearchBar";
 import { FilterAndSortPopUp } from "./FilterAndSortPopUp/FilterAndSortPopUp";
+import { RequiresAuth } from "./RequiresAuth/RequiresAuth";
 
-export { RoutesComponent, Header, SideBar, CreateNote, NotesListing, Note, LabeledNotes, ArchivedNotes, DeletedNotes, SearchBar, FilterAndSortPopUp };
\ No newline at end of file
+export { RoutesComponent, Header, SideBar, CreateNote, NotesListing, Note, LabeledNotes, ArchivedNotes, DeletedNotes, SearchBar, FilterAndSortPopUp, RequiresAuth };
```
Comment (Kothalkarpriya) [inline src/components/index.js:14]: you can use code formatter

**Review by Kothalkarpriya — APPROVED** [review]
Code is clean just use formatter for each page, it will be more clear to understand

File: src/pages/Authentication/LogOut/LogOut.js [hunk 2]
```diff
@@ -1,17 +1,25 @@
 import "./logOut.css";
 import { Link } from "react-router-dom";
+import { useAuth } from "../../../contexts";
 
 const LogOutPage = () => {
+
+    const { setIsUserLoggedIn } = useAuth();
```
Comment (t007rushi) [inline src/pages/Authentication/LogOut/LogOut.js:7]: never pass setter instead create function and pass as callback

File: src/pages/Authentication/LogIn/LogIn.js [hunk 3]
```diff
@@ -1,21 +1,34 @@
...
-                <input type="password" placeholder="Password" id="tn_login-password" />
+                    <label htmlFor="tn_login-password">Password</label>
+                    <input type="password" placeholder="Password" id="tn_login-password" value={formData.userPwd} onChange={(e) => {setLogInErrMsg(""); setFormData({...formData, userPwd: e.target.value})}} required />
 
-                <button className="et_p-simple-btn primary-color btn">LogIn</button>
+                    <button type="submit" className="et_p-simple-btn primary-color btn" onClick={(e) => loginHandler(e)}>LogIn</button>
+                </form>
+                <button className="et_p-simple-btn primary-color btn" onClick={() => handleGuestLogin()}>LogIn As A Guest</button>
+                <div>
+                    <p onClick={() => navigate("/SignUp")}>Create new account:)</p>
```
Comment (t007rushi) [inline src/pages/Authentication/LogIn/LogIn.js:27]: no ```click``` event, use ```Link``` instead

File: src/contexts/auth-context.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    const [ signUpErrMsg, setSignUpErrMsg ] = useState("");
+    const [ logInErrMsg, setLogInErrMsg ] = useState("");
+    const [ formData, setFormData ] = useState({ userEmail: "", userPwd: "" })
+    const navigate = useNavigate();
+    const location = useLocation();
+
+    const signUpHandler = async (e) => {
+        e.preventDefault();
+        if( formData.userEmail!=="" && formData.userPwd!==""){
+            if(/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(formData.userEmail)){
```
Comment (t007rushi) [inline src/contexts/auth-context.js:19]: nicely done

**Review by t007rushi — APPROVED** [review]
host it else good to go

File: src/pages/Authentication/LogIn/LogIn.js (hunk 3 above)
Comment (Namrata-J) [reply src/pages/Authentication/LogIn/LogIn.js:27]: okay will take for it from now!

**Pull request merged**